node-graph/gmath-nodes/src/lib.rs

```diff
@@ -0,0 +1,532 @@
+		max
+	} else {
+		value
+	}
+}
+
+/// The equality operation (==) compares two values and returns true if they are equal, or false if they are not.
+#[node_macro::node(category("Math: Logic"))]
+fn equals<U: std::cmp::PartialEq<T>, T>(
+	_: impl Ctx,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32, DVec2, &DVec2, &str)] value: T,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32, DVec2, &DVec2, &str)] other_value: U,
+) -> bool {
+	other_value == value
+}
+
+/// The inequality operation (!=) compares two values and returns true if they are not equal, or false if they are.
+#[node_macro::node(category("Math: Logic"))]
+fn not_equals<U: std::cmp::PartialEq<T>, T>(
+	_: impl Ctx,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32, DVec2, &DVec2, &str)] value: T,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32, DVec2, &DVec2, &str)] other_value: U,
+) -> bool {
+	other_value != value
+}
+
+/// The less-than operation (<) compares two values and returns true if the first value is less than the second, or false if it is not.
+/// If enabled with "Or Equal", the less-than-or-equal operation (<=) will be used instead.
+#[node_macro::node(category("Math: Logic"))]
+fn less_than<T: std::cmp::PartialOrd<T>>(
+	_: impl Ctx,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32)] value: T,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32)] other_value: T,
+	or_equal: bool,
+) -> bool {
+	if or_equal { value <= other_value } else { value < other_value }
+}
+
+/// The greater-than operation (>) compares two values and returns true if the first value is greater than the second, or false if it is not.
+/// If enabled with "Or Equal", the greater-than-or-equal operation (>=) will be used instead.
+#[node_macro::node(category("Math: Logic"))]
+fn greater_than<T: std::cmp::PartialOrd<T>>(
+	_: impl Ctx,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32)] value: T,
+	#[implementations(f64, &f64, f32, &f32, u32, &u32)] other_value: T,
+	or_equal: bool,
+) -> bool {
+	if or_equal { value >= other_value } else { value > other_value }
+}
+
+/// The logical or operation (||) returns true if either of the two inputs are true, or false if both are false.
+#[node_macro::node(category("Math: Logic"))]
+fn logical_or(_: impl Ctx, value: bool, other_value: bool) -> bool {
+	value || other_value
+}
+
+/// The logical and operation (&&) returns true if both of the two inputs are true, or false if any are false.
+#[node_macro::node(category("Math: Logic"))]
+fn logical_and(_: impl Ctx, value: bool, other_value: bool) -> bool {
+	value && other_value
+}
+
+/// The logical not operation (!) reverses true and false value of the input.
+#[node_macro::node(category("Math: Logic"))]
+fn logical_not(_: impl Ctx, input: bool) -> bool {
+	!input
+}
+
+/// Constructs a bool value which may be set to true or false.
+#[node_macro::node(category("Value"))]
+fn bool_value(_: impl Ctx, _primary: (), #[name("Bool")] bool_value: bool) -> bool {
+	bool_value
+}
+
```

Comment:
I'm wondering if it would make more sense to move these to the logic mod now that the mod is no longer called ops

Reply:
We can split up the lib into smaller mods, though I'd create a new logic op and not put it into the old, since the old is almost exclusively string manipulation except for SwitchNode (which I may move here)

Reply:
Alright, just wanted to make you aware of this, we can always move things later once we are a bit more settled into the new structure